### PR TITLE
Adding unsigned load/stores

### LINKED ISSUE #94: manual does not mention unsigned load insn,  l{b|h|w}u
The manual lists signed loads, l{b|h|w|d}  but not the unsigned variant, l{b|h|w}u.  Suggest adding it.

### PR DESCRIPTION
**Pull request by cmuellner**
We provide a range of pseudoinstructions to access globals. Currently only the signed loads/stores are listed. Let's add the unsigned accesses as well.

Fixes #94.

### CODE REVIEW

File: riscv-asm.md [hunk 1]
```diff
@@ -693,8 +693,8 @@ Load and Store Global
 The following pseudoinstructions are available to load from and store to
 global objects:
 
-  * `l{b|h|w|d} <rd>, <symbol>`: load byte, half word, word or double word from global[^1]
-  * `s{b|h|w|d} <rd>, <symbol>, <rt>`: store byte, half word, word or double word to global[^2]
+  * `l{b|bu|h|hu|w|wu|d} <rd>, <symbol>`: load byte, half word, word or double word from global[^1]
+  * `s{b|bu|h|hu|w|wu|d} <rd>, <symbol>, <rt>`: store byte, half word, word or double word to global[^2]
```
Comment (aswaterman) [inline riscv-asm.md:697]: There are no unsigned stores (just loads).

**Review by kito-cheng — APPROVED** [review]

File: riscv-asm.md [hunk 2]
```diff
@@ -693,8 +693,8 @@ Load and Store Global
 The following pseudoinstructions are available to load from and store to
 global objects:
 
-  * `l{b|h|w|d} <rd>, <symbol>`: load byte, half word, word or double word from global[^1]
-  * `s{b|h|w|d} <rd>, <symbol>, <rt>`: store byte, half word, word or double word to global[^2]
+  * `l{b|bu|h|hu|w|wu|d} <rd>, <symbol>`: load byte, half word, word or double word from global[^1]
```
Comment (aswaterman) [inline riscv-asm.md:696]: Consider changing the description to "load unsigned byte, half word, word or double word from global"?

File: riscv-asm.md [hunk 3]
```diff
@@ -912,8 +912,8 @@ la rd, symbol                | auipc rd, symbol[31:12]; addi rd, rd, symbol[11:0
 la rd, symbol                | auipc rd, symbol@GOT[31:12]; l{w\|d} rd, symbol@GOT[11:0]\(rd\) | Load address | With `.option pic`
 lla rd, symbol               | auipc rd, symbol[31:12]; addi rd, rd, symbol[11:0]            | Load local address
 lga rd, symbol               | auipc rd, symbol@GOT[31:12]; l{w\|d} rd, symbol@GOT[11:0]\(rd\) | Load global address
-l{b\|h\|w\|d} rd, symbol     | auipc rd, symbol[31:12]; l{b\|h\|w\|d} rd, symbol[11:0]\(rd\) | Load global
-s{b\|h\|w\|d} rd, symbol, rt | auipc rt, symbol[31:12]; s{b\|h\|w\|d} rd, symbol[11:0]\(rt\) | Store global
+l{b\|bu\|h\|hu\|w\|wu\|d} rd, symbol     | auipc rd, symbol[31:12]; l{b\|h\|w\|d} rd, symbol[11:0]\(rd\) | Load global
+s{b\|bu\|h\|hu\|w\|wu\|d} rd, symbol, rt | auipc rt, symbol[31:12]; s{b\|h\|w\|d} rd, symbol[11:0]\(rt\) | Store global
```
Comment (aswaterman) [inline riscv-asm.md:916]: Delete

File: riscv-asm.md (hunk 1 above)
Comment (cmuellner) [reply riscv-asm.md]: Yes, sorry for this.

File: riscv-asm.md (hunk 2 above)
Comment (cmuellner) [reply riscv-asm.md]: I've added a new line for the unsigned loads and used this description (without double word).

File: riscv-asm.md [hunk 4]
```diff
@@ -912,8 +912,8 @@ la rd, symbol                | auipc rd, symbol[31:12]; addi rd, rd, symbol[11:0
 la rd, symbol                | auipc rd, symbol@GOT[31:12]; l{w\|d} rd, symbol@GOT[11:0]\(rd\) | Load address | With `.option pic`
 lla rd, symbol               | auipc rd, symbol[31:12]; addi rd, rd, symbol[11:0]            | Load local address
 lga rd, symbol               | auipc rd, symbol@GOT[31:12]; l{w\|d} rd, symbol@GOT[11:0]\(rd\) | Load global address
-l{b\|h\|w\|d} rd, symbol     | auipc rd, symbol[31:12]; l{b\|h\|w\|d} rd, symbol[11:0]\(rd\) | Load global
-s{b\|h\|w\|d} rd, symbol, rt | auipc rt, symbol[31:12]; s{b\|h\|w\|d} rd, symbol[11:0]\(rt\) | Store global
+l{b\|bu\|h\|hu\|w\|wu\|d} rd, symbol     | auipc rd, symbol[31:12]; l{b\|h\|w\|d} rd, symbol[11:0]\(rd\) | Load global
```
[the remark this answers is not in the document]
Comment (cmuellner) [reply riscv-asm.md]: I've moved the unsigned loads in their own line and use the description that you've mentioned.

File: riscv-asm.md (hunk 3 above)
Comment (cmuellner) [reply riscv-asm.md]: Done.

**Review by aswaterman — APPROVED** [review]
LGTM!

**Pull request merged**